cpp/src/ffi_go/ir/deserializer.cpp

```diff
@@ -0,0 +1,308 @@
+        Deserializer* deserializer{static_cast<Deserializer*>(ir_deserializer)};
+
+        IRErrorCode err{};
+        epoch_time_ms_t timestamp{};
+        if constexpr (std::is_same_v<encoded_variable_t, eight_byte_encoded_variable_t>) {
+            err = eight_byte_encoding::decode_next_message(
+                    ir_buf,
+                    deserializer->m_log_event.m_log_message,
+                    timestamp
+            );
+        } else if constexpr (std::is_same_v<encoded_variable_t, four_byte_encoded_variable_t>) {
+            epoch_time_ms_t timestamp_delta{};
+            err = four_byte_encoding::decode_next_message(
+                    ir_buf,
+                    deserializer->m_log_event.m_log_message,
+                    timestamp_delta
+            );
+            timestamp = deserializer->m_timestamp + timestamp_delta;
+        } else {
+            static_assert(cAlwaysFalse<encoded_variable_t>, "Invalid/unhandled encoding type");
+        }
+        if (IRErrorCode_Success != err) {
+            return static_cast<int>(err);
+        }
+        deserializer->m_timestamp = timestamp;
+
+        size_t pos{0};
+        if (ErrorCode_Success != ir_buf.try_get_pos(pos)) {
+            return static_cast<int>(IRErrorCode_Decode_Error);
+        }
+        *ir_pos = pos;
+        log_event->m_log_message.m_data = deserializer->m_log_event.m_log_message.data();
+        log_event->m_log_message.m_size = deserializer->m_log_event.m_log_message.size();
+        log_event->m_timestamp = deserializer->m_timestamp;
+        return static_cast<int>(IRErrorCode_Success);
+    }
+
+    template <class encoded_variable_t>
+    auto deserialize_wildcard_match(
+            ByteSpan ir_view,
+            void* ir_deserializer,
+            TimestampInterval time_interval,
+            MergedWildcardQueryView merged_query,
+            size_t* ir_pos,
+            LogEventView* log_event,
+            size_t* matching_query
+    ) -> int {
+        BufferReader ir_buf{static_cast<char const*>(ir_view.m_data), ir_view.m_size};
+        Deserializer* deserializer{static_cast<Deserializer*>(ir_deserializer)};
+        std::string_view const query_view{
+                merged_query.m_queries.m_data,
+                merged_query.m_queries.m_size};
+        std::span<size_t> const end_offsets{
+                merged_query.m_end_offsets.m_data,
+                merged_query.m_end_offsets.m_size};
+        std::span<bool> const case_sensitivity{
+                merged_query.m_case_sensitivity.m_data,
+                merged_query.m_case_sensitivity.m_size};
+
+        std::vector<std::pair<std::string_view, bool>> queries(merged_query.m_end_offsets.m_size);
+        size_t pos{0};
+        for (size_t i{0}; i < merged_query.m_end_offsets.m_size; i++) {
+            queries[i].first = query_view.substr(pos, end_offsets[i]);
+            queries[i].second = case_sensitivity[i];
```

Comment:
```suggestion
            auto& [query_str_view, is_case_sensitive]{queries[i]};
            query_str_view = query_view.substr(pos, end_offsets[i]);
            is_case_sensitive = case_sensitivity[i];
```
How about refactoring like this for better readability?